pkg/vhdbuilder/datamodel/component_configs.go

```diff
@@ -124,41 +123,44 @@ func NewKubeProxyImagesFromFile(path string) (*KubeProxyImages, error) {
 	return ret, nil
 }
 
+// processProxyImages is a helper function for ToImageList().
+func processProxyImages(image *ContainerImage, ret *[]string) error {
+	var err error
+	var amd64OnlyImageList []string
+	var multiArchImageList []string
+
+	if image.Amd64OnlyVersions != nil {
+		amd64OnlyImageList, err = toImageList(image.DownloadURL, image.Amd64OnlyVersions)
+		*ret = append(*ret, amd64OnlyImageList...)
+	}
+
+	if image.MultiArchVersions != nil {
+		multiArchImageList, err = toImageList(image.DownloadURL, image.MultiArchVersions)
+		*ret = append(*ret, multiArchImageList...)
+	}
+	return err
+}
+
 /*
 ToImageList returns a list of image names from the given KubeProxyImages object.
 The image names are generated from the given KubeProxyImages object and the given downloadURL.
 The image names are generated by replacing the * in the downloadURL with the image tag.
 */
-func (k *KubeProxyImages) ToImageList() []string {
-	ret := []string{}
+func (k *KubeProxyImages) ToImageList() ([]string, error) {
+	var err error
+	ret := &[]string{}
 
 	if k.DockerKubeProxyImages != nil && k.DockerKubeProxyImages.ContainerImages != nil {
 		for _, image := range k.DockerKubeProxyImages.ContainerImages {
-			if image.Amd64OnlyVersions != nil {
-				amd64OnlyImageList, _ := toImageList(image.DownloadURL, image.Amd64OnlyVersions)
-				ret = append(ret, amd64OnlyImageList...)
-			}
-
-			if image.MultiArchVersions != nil {
-				multiArchImageList, _ := toImageList(image.DownloadURL, image.MultiArchVersions)
-				ret = append(ret, multiArchImageList...)
-			}
+			err = processProxyImages(image, ret)
```

Comment:
I'd either return this error immediately or build up a cumulative error and return that - as this stands right now all errors except the last will be overwritten right?